lib/std/heap.zig

```diff
@@ -719,6 +719,54 @@ test "Thread safe FixedBufferAllocator" {
     try testAllocatorAlignedShrink(fixed_buffer_allocator.threadSafeAllocator());
 }
 
+test "FixedBufferAllocator at comptime" {
+    comptime {
+        var buf: [16]u8 = undefined;
+        var fba = FixedBufferAllocator.init(&buf);
+
+        {
+            var slice = fba.allocator().alloc(u8, 8) catch unreachable;
+            slice[3] = 2;
+            assert(slice[3] == 2);
+            assert(fba.allocator().resize(slice, 16));
+            slice.len = 16;
+            slice[15] = 5;
+            assert(slice[15] == 5);
+            assert(fba.allocator().resize(slice, 4));
+            slice.len = 4;
+            assert(slice[3] == 2);
+            fba.allocator().free(slice);
+        }
+        assert(fba.end_index == 0);
+
+        {
+            var slice = fba.allocator().alloc(u32, 4) catch unreachable;
+            slice[0] = 20;
+            slice[3] = 2;
+            assert(slice[3] == 2);
+            assert(fba.allocator().resize(slice, 1));
+            slice.len = 1;
+            assert(slice[0] == 20);
+            assert(fba.allocator().resize(slice, 0));
+            slice.len = 0;
+            fba.allocator().free(slice);
+        }
+        assert(fba.end_index == 0);
+
+        {
+            const T = extern struct {
```

Comment:
Note that removing `extern` breaks this with:
```
/home/josh/tmp/asdf/a.zig:46:32: error: comptime mutation of a reinterpreted pointer requires type 'a.test.FixedBufferAllocator at comptime__struct_1720' to have a well-defined memory layout
            slice[3] = .{ .a = 1, .b = 2 };
                               ^
```
And bizarrely turning this into `packed` makes the below assertion fail. Looking at `compileLog`s indicates that there seems to be a bug with assigning packed structs into an array at comptime, or else i'm misunderstanding what a packed struct is. but that can be a separate bug report.